cranelift-codegen/meta/src/gen_encodings.rs

```diff
@@ -78,7 +78,7 @@ use crate::unique_table::UniqueSeqTable;
 fn emit_instp(instp: &InstructionPredicate, has_func: bool, fmt: &mut Formatter) {
     if instp.is_type_predicate() {
         fmt.line("let args = inst.arguments(&func.dfg.value_lists);");
-        fmt.line(instp.rust_predicate());
+        fmt.line(instp.rust_predicate("func").unwrap());
```

Comment:
It would mean publicizing the node, which was prevented in previous code. However, I can morph "is_type_predicate" into "type_predicate(func_str: &str) -> Option<String>", with the same effect and nice typesystem-ness.